omnipath/_misc/dtypes.py

```diff
@@ -0,0 +1,175 @@
+from typing import Union, Iterable
+
+import pandas as pd
+import numpy as np
+
+
+TRUE = frozenset(('true', 't', 'yes', 'y'))
+FALSE = frozenset(('false', 'f', 'no', 'n'))
+BOOL = frozenset().union(TRUE, FALSE)
+NA = frozenset((
+    'na', 'NA', 'NaN',
+    'none', 'None', None,
+    pd.NA, pd.NaT, np.NAN, np.nan
+))
+INT = ('int64', 'uint64')
+ALL = INT + ('float64', 'string')
+
+
+def auto_dtype(
+    data: Union[pd.DataFrame, pd.Series, Iterable],
+    convert: bool = True,
+    categories: bool = True,
+    **kwargs,
+) -> Union[pd.DataFrame, pd.Series, str, dict]:
+    """
+    Automatically guesses and optionally converts data types of a dataframe,
+    series or other iterable.
+
+    Parameters
+    ----------
+    data
+        A dataframe or an array like object such as :class:`pandas.Series`,
+        :class:`numpy.ndarray` or list.
+    convert
+        Convert the data types to the best fitting types.
+    categories
+        Use the `category` data type for string variables with a small
+        number of values compared to their size.
+    **kwargs
+        For dataframes, manually set the desired data type of certain
+        variables.
+
+    Returns
+    -------
+    :class:`pandas.DataFrame` or :class:`pandas.Series` or str or list
+        If :arg:`convert` is `True`: a dataframe or series with its data
```

Comment:
Please also remove the `:arg:` - although docstrings are not being rendered for this function, it's just for safety/future-proofness.